src/main/java/com/atlassian/db/replica/spi/ConnectionProvider.java

```diff
@@ -3,17 +3,12 @@
 import java.sql.Connection;
 import java.sql.SQLException;
 
-public interface ConnectionProvider {
+public interface ConnectionProvider extends ReplicaConnectionProvider {
 
     boolean isReplicaAvailable();
 
     /**
-     * @return a connection to the main databsase
+     * @return a connection to the main database
      */
     Connection getMainConnection() throws SQLException;
-
-    /**
-     * @return a connection to a replica database
-     */
-    Connection getReplicaConnection() throws SQLException;
```

Comment:
The change is not compatible.

Reply:
I see ReplicaConnectionProvider takes it.

Reply:
What's the reason to change it now?

Reply:
I extracted this method into `ReplicaConnectionProvider` interface to reuse it in `ReplicaConnectionPerUrlProvider` because as per @dwatl's suggestion we shouldn't make the lib connection-pool-aware